build.gradle

```diff
@@ -1,109 +1,147 @@
+                }
+            }
+        }
+    }
 }
 
-dependencies {
-    compile group: 'org.controlsfx', name: 'controlsfx', version: '8.40.10'
-    compile group: 'com.google.guava', name: 'guava', version: '18.0'
-    compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.4'
-    compile group: 'org.python', name: 'jython', version: '2.7.0'
-    compile group: 'com.thoughtworks.xstream', name: 'xstream', version: '1.4.8'
-    compile group: 'org.bytedeco', name: 'javacv', version: '1.1'
-    compile group: 'org.bytedeco.javacpp-presets', name: 'opencv', version: '3.0.0-1.1'
-    compile group: 'org.bytedeco.javacpp-presets', name: 'opencv', version: '3.0.0-1.1', classifier: osdetector.classifier.replace("osx", "macosx")
-
-    testCompile group: 'org.hamcrest', name: 'hamcrest-all', version: '1.3'
-    testCompile group: 'junit', name: 'junit', version: '4.12'
-    testCompile group: 'org.testfx', name: 'testfx-core', version: '4.0.+'
-    testCompile group: 'org.testfx', name: 'testfx-junit', version: '4.0.+'
-}
+project(":core") {
+    apply plugin: 'java'
+    apply plugin: "com.github.johnrengelman.shadow"
 
-compileJava.source sourceSets.generated.java, sourceSets.main.java
+    configurations {
+        shadow
+    }
 
-task generateCodeFromSource(type: CodeGenerator) {
-    description 'Generates the Operation Wrappers for the OpenCV methods.\n To skip run with parm `-PskipGenerate`'
-    dest sourceSets.generated.java.srcDirs
-    removeExisting true
-}
-generateCodeFromSource.onlyIf { !project.hasProperty('skipGenerate') }
-compileJava.dependsOn generateCodeFromSource
-
-// Turn on test results
-test {
-    testLogging {
-        afterSuite { desc, result ->
-            if (!desc.parent && project.hasProperty('printTestResults')) {
-                println "Test results: ${result.resultType} (${result.testCount} tests, ${result.successfulTestCount} passed, ${result.failedTestCount} failed, ${result.skippedTestCount} skipped)"
-            }
-        }
+    dependencies {
+        compile group: 'com.google.guava', name: 'guava', version: '18.0'
+        compile group: 'org.bytedeco', name: 'javacv', version: '1.1'
+        compile group: 'org.bytedeco.javacpp-presets', name: 'opencv', version: '3.0.0-1.1'
+        compile group: 'org.python', name: 'jython', version: '2.7.0'
+        compile group: 'com.thoughtworks.xstream', name: 'xstream', version: '1.4.8'
+        testCompile group: 'org.bytedeco.javacpp-presets', name: 'opencv', version: '3.0.0-1.1', classifier: osdetector.classifier.replace("osx", "macosx")
     }
-}
 
-jacocoTestReport {
-    reports {
-        xml.enabled = true
-        html.enabled = true
+    mainClassName = 'edu.wpi.grip.core.Main'
+
+    jar {
+        manifest {
+            attributes 'Implementation-Title': 'GRIP CORE', 'Implementation-Version': version, 'Main-Class': mainClassName
```

Comment:
Just following this: https://github.com/johnrengelman/shadow#configure-manifest-file
Do you know?